tests/TestCase.php

```diff
@@ -11,8 +11,18 @@
 
     protected $baseUrl = 'https://dnsrecords.io.dev';
 
-    protected function sendCommand(string $command): TestResponse
+    protected function sendCommand(string $command, string $url = null): TestResponse
     {
-        return $this->post("{$this->baseUrl}/{$command}", compact('command'));
+        $url = $url ? $this->baseUrl . $url : "{$this->baseUrl}/{$command}";
+
+        return $this->post($url, compact('command'));
+    }
+
+    protected function getFlashMessage(): ?string
+    {
+        $flash = app('session.store')
+            ->get('flash_notification');
+
+        return $flash ? $flash->first()->message : null;
```

Comment:
If you always return a string you can drop the nullable type hint. I prefer this when null and an empty value of the type have the same meaning in the context.

Reply:
To me that's a different meaning: `null` means no flash message is set, while an empty string could also mean an empty flash message was set.